### ISSUE TITLE
Improve accessibility of Newsletter Subscribe form

### ISSUE THREAD

**Issue by danieltott** [Virtual-Coffee/virtualcoffee.io#447]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Context

There are a couple issues to fix with the newsletter subscribe form, found on https://virtualcoffee.io/newsletter/ (and individual newsletter issues).

The code that generates this form can be found here: https://github.com/Virtual-Coffee/virtualcoffee.io/blob/main/src/_includes/macros/newslettersub.njk

Running a WAVE test will show three issues:

![image](https://user-images.githubusercontent.com/360261/138116508-0166ac5d-3a0f-45df-99b6-cf313ddadd0e.png)

- Missing form label
- Broken ARIA reference
- Fieldset missing legend

### Form label:

The form includes a [honey pot](https://www.thryv.com/blog/honeypot-technique/) field, but this field isn't labelled. The idea is that the field should be empty, but spam bots will be tricked into filling it out. We should add a label that is visually hidden, but has the text "please leave this field empty" or something like that.

### Broken ARIA reference:

This is simply a typo:

https://github.com/Virtual-Coffee/virtualcoffee.io/blob/8fc7a906be9a23898d2f98fb8883d78c06d148de/src/_includes/macros/newslettersub.njk#L13-L21

`aria-describedby` is set to `emailHelp`, but the id of the help content is `nameHelp`. They should match.

### Fieldset missing legend:

From the WAVE tool:

> If a higher level description is necessary for the user to understand the function or purpose of the controls within the fieldset, provide this description within the <legend>. If this description or grouping is not necessary, the fieldset should probably be removed. 

In this case, we don't need a `fieldset`, and it can be safely removed.

- [ ] Add label to honeypot field
- [ ] Fix `aria-describedby` value
- [ ] Remove `fieldset`

### Code of Conduct

- [X] I've read the Code of Conduct and understand my responsibilities as a member of the Virtual Coffee community

**Comment by adiati98** [comment]
Hi @danieltott,
May I try this one? 😊
